src/gcodePlanner.cpp

```diff
@@ -448,6 +448,80 @@ void GCodePlanner::addLinesByOptimizer(Polygons& polygons, GCodePathConfig* conf
     }
 }
 
+int GCodePlanner::spiralizeWallSlice(GCodePathConfig* config, PolygonRef wall, PolygonRef last_wall, int last_wall_seam_vertex_idx)
+{
+    // The spiral has to continue on in an anti-clockwise direction from where the last layer finished, it can't jump backwards
+
+    const int n_points = wall.size();
+
+    // seam_vertex_idx is going to be the index of the seam vertex in the current wall polygon
+    // initially we choose the vertex that is closest to the seam vertex in the last spiralized layer processed
+
+    int seam_vertex_idx = PolygonUtils::findClosest((last_wall_seam_vertex_idx < 0) ? lastPosition : last_wall[last_wall_seam_vertex_idx], wall).point_idx;
+
+    if (last_wall_seam_vertex_idx >= 0)
+    {
+        // now we check that the vertex following the seam vertex is not to the right of the seam vertex in the last layer
+        // and if it is we move forward
+
+        // get the inward normal of the last layer seam vertex
+        Point last_wall_seam_vertex = last_wall[last_wall_seam_vertex_idx];
+        Point last_wall_seam_vertex_inward_normal = PolygonUtils::getVertexInwardNormal(last_wall, last_wall_seam_vertex_idx);
+
+        // create a vector from the normal so that we can then test the vertex following the candidate seam vertex to make sure it is on the correct side
+        Point last_wall_seam_vertex_vector = last_wall_seam_vertex + last_wall_seam_vertex_inward_normal;
+
+        // now test the vertex following the candidate seam vertex and if it lies to the left or on the vector, it's good to use
+        const int first_seam_vertex_idx = seam_vertex_idx;
+        float a = LinearAlg2D::getAngleLeft(last_wall_seam_vertex_vector, last_wall_seam_vertex, wall[(seam_vertex_idx + 1) % n_points]);
+        //std::cerr << layer_nr << ": angle = " << a * 180/M_PI << "\n";
+
+        while (a > M_PI)
+        {
+            // the vertex was on the right of the vector so move the seam vertex on
+            seam_vertex_idx = (seam_vertex_idx + 1) % n_points;
+            a = LinearAlg2D::getAngleLeft(last_wall_seam_vertex_vector, last_wall_seam_vertex, wall[(seam_vertex_idx + 1) % n_points]);
+            //std::cerr << "new angle = " << a * 180/M_PI << "\n";
+
+            if (seam_vertex_idx == first_seam_vertex_idx)
+            {
+                // this shouldn't happen!
+                break;
+            }
+        }
+    }
+
+    Polygons last_wall_polygons;
+    last_wall_polygons.add(last_wall);
+    const int max_dist = MM2INT(config->getLineWidth() * 5);
+    // extrude to the points following the seam vertex
+    // the last point is the seam vertex as the polygon is a loop
+    for (int i = 1; i <= n_points; ++i)
+    {
+        // p is a point from the current wall polygon
+        const Point& p = wall[(seam_vertex_idx + i) % n_points];
```

Comment:
Please don't use abbreviations in variable names.
I myself only abbreviate 'index' to 'idx' and 'polygon' to 'poly', because these are used everywhere in the code.
Other variables are better written out fully, to make the code more reabible.